_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,983 @@
+      "ml_commons": {
+        "only_run_on_ml_node": "false",
+        "model_access_control_enabled": "true",
+        "native_memory_threshold": "99"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+#### Advanced
+
+For a more advanced setup, note the following requirements:
+
+- To register a custom model, you need to specify an additional `"allow_registering_model_via_url": "true"` cluster setting. 
+- On clusters with dedicated ML nodes, you may want to specify `"only_run_on_ml_node": "true"` for improved performance. 
+
+For more information about ML-related cluster settings, see [ML Commons cluster settings]({{site.url}}{{site.baseurl}}/ml-commons-plugin/cluster-settings/).
+
+## Tutorial overview
+
+This tutorial consists of the following steps:
+
+1. [**Set up an ML language model**](#step-1-set-up-an-ml-language-model).
+    1. [Choose a language model](#step-1a-choose-a-language-model).
+    1. [Register a model group](#step-1b-register-a-model-group).
+    1. [Register the model to the model group](#step-1c-register-the-model-to-the-model-group).
+    1. [Deploy the model](#step-1d-deploy-the-model).
+1. [**Ingest data with neural search**](#step-2-ingest-data-with-neural-search).
+    1. [Create an ingest pipeline for neural search](#step-2a-create-an-ingest-pipeline-for-neural-search).
+    1. [Create a k-NN index](#step-2b-create-a-k-nn-index).
+    1. [Ingest documents into the index](#step-2c-ingest-documents-into-the-index).
+1. [**Search the data**](#step-3-search-the-data).
+   - [Search using a keyword search](#search-using-a-keyword-search).
+   - [Search using a neural search](#search-using-a-neural-search).
+   - [Search using a combined keyword search and neural search](#search-using-a-combined-keyword-search-and-neural-search).
+
+Some steps in the tutorial contain optional `Test it` sections. You can ensure that the step was successful by running requests in these sections.
+
+After you're done, follow the steps in the [Clean up](#clean-up) section to delete all created components.
+
+## Tutorial
+
+You can follow this tutorial using your command line or the OpenSearch Dashboards [Dev Tools console]({{site.url}}{{site.baseurl}}/dashboards/dev-tools/run-queries/).
+
+## Step 1: Set up an ML language model
+
+Neural search requires a language model in order to generate vector embeddings from text fields, both at ingestion time and query time.
+
+### Step 1(a): Choose a language model
+
+For this tutorial, you'll use the [DistilBERT](https://huggingface.co/docs/transformers/model_doc/distilbert) model from Hugging Face. It is one of the pretrained sentence transformer models available in OpenSearch. You'll need the name, version, and dimension of the model to register it. You can find this information in the [pretrained model table]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#sentence-transformers) by selecting the `config_url` link corresponding to the model's TorchScript artifact:
```

Comment:
If we have any reason this model was picked we should call that out here. That way others can find a model that suites them.

Reply:
For purpose of testing Hybrid search I was following [previous blog ](https://opensearch.org/blog/semantic-search-solutions/)from science team on building semantic search, they have mentioned TAS-B as a showing one of the best results among other models. Plus it's supported as one of pre-trained models by our ml-commons https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/#sentence-transformers